File: 1.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
  int64_t S=0, cnt=1,n,k=0;
   cin >> n;
    vector<int64_t> A(n*(n-1)/2);
    vector<int> x(n),y(n);
     for(int i=0; i<n; i++)
     {
         cin >> x[i] >> y[i];
     }
      for(int i=0; i<n-1; i++)
      {
        for(int j=i+1; j<n; j++)
        { 
         int64_t dx = x[i]-x[j];
         int64_t dy = y[i]-y[j];
          A[k++] = ((dx*dx+dy*dy)<<32) + ((x[i]+x[j])<<16) + y[i]+y[j];
        }
      }
        sort(A.begin(),A.end());
         for(int i=1; i<A.size(); i++)
         {
             if (A[i]==A[i-1])
             {
                 cnt++;
             }
             else
             {
                S += cnt*(cnt-1)/2;  
                 cnt=1;
             }
         }
          S += cnt*(cnt-1)/2;
           cout << S;
            return 0;
}

File: 1.js
let input = '';
 process.stdin.on('data', 
                 ////----------------------------
                 (chunk) => {
                             input += chunk;
                 });

 process.stdin.on('end', 
                  ////--------------------------- 
                  () => {
                     const lines = input.trim().split('\n');  
                      const n = parseInt(lines[0], 10);        
                     const x = [];
                     const y = [];

                       for (let i = 1; i <= n; i++) 
                       {
                           const [xi, yi] = lines[i].split(' ').map(Number);  
                            x.push(xi);
                            y.push(yi);
                       }
                     let S = 0;
                     let cnt = 1;
                     let k = 0;
                     const A = [];

                        for (let i = 0; i < n - 1; i++) 
                        {
                          for (let j = i + 1; j < n; j++) 
                          {
                            const dx = x[i] - x[j];
                            const dy = y[i] - y[j];
                             A[k++] = ((dx * dx + dy * dy) << 32) + ((x[i] + x[j]) << 16) + y[i] + y[j];
                          }
                       }
                         A.sort((a, b) => a - b); 
                          for (let i = 1; i < A.length; i++) 
                          {
                           if (A[i] === A[i - 1]) 
                           {
                            cnt++;
                           } 
                           else 
                              {
                               S += cnt * (cnt - 1) / 2;
                                cnt = 1;
                              }
                         }
                          S += cnt * (cnt - 1) / 2;
                           console.log(S);
                        });
